src/content_working_group/lib.rs

```diff
@@ -0,0 +1,2422 @@
+
+            // Add application to set of application in curator opening
+            CuratorOpeningById::<T>::mutate(curator_opening_id, |curator_opening| {
+                curator_opening.curator_applications.insert(new_curator_application_id);
+            });
+
+            // Trigger event
+            Self::deposit_event(RawEvent::AppliedOnCuratorOpening(curator_opening_id, new_curator_application_id));
+        }
+
+        /// An active curator can update the associated role account.
+        pub fn update_curator_role_account(
+            origin,
+            member_id: T::MemberId,
+            curator_id: CuratorId<T>,
+            new_role_account: T::AccountId
+        ) {
+            // Ensure that origin is signed by member with given id.
+            ensure_on_wrapped_error!(
+                members::Module::<T>::ensure_member_controller_account_signed(origin, &member_id)
+            )?;
+
+            // Ensure that member is this curator
+            let actor_in_role = role_types::ActorInRole::new(role_types::Role::Curator, curator_id);
+
+            ensure!(
+                members::MembershipIdByActorInRole::<T>::get(actor_in_role) == member_id,
+                MSG_CURATOR_NOT_CONTROLLED_BY_MEMBER
+            );
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Update role account
+            CuratorById::<T>::mutate(curator_id, |curator| {
+                curator.role_account = new_role_account.clone()
+            });
+
+            // Trigger event
+            Self::deposit_event(RawEvent::CuratorRoleAccountUpdated(curator_id, new_role_account));
+        }
+
+        /// An active curator can update the reward account associated
+        /// with a set reward relationship.
+        pub fn update_curator_reward_account(
+            origin,
+            curator_id: CuratorId<T>,
+            new_reward_account: T::AccountId
+        ) {
+
+            // Ensure there is a signer which matches role account of curator corresponding to provided id.
+            let curator = Self::ensure_active_curator_signed(origin, &curator_id)?;
+
+            // Ensure the curator actually has a recurring reward
+            let relationship_id = Self::ensure_curator_has_recurring_reward(&curator)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Update, only, reward account.
+            recurringrewards::Module::<T>::set_reward_relationship(
+                relationship_id,
+                Some(new_reward_account.clone()), // new_account
+                None, // new_payout
+                None, //new_next_payment_at
+                None //new_payout_interval
+            )
+            .expect("Must be set, since curator has recurring reward");
+
+            // Trigger event
+            Self::deposit_event(RawEvent::CuratorRewardAccountUpdated(curator_id, new_reward_account));
+        }
+
+        /// An active curator leaves role
+
```

Comment:
This can be called via sudo extrinsic. I don't think this is what you intended. Its probably the method that will be called from handler for StakingEventHandler::unstaked ?

Reply:
No values are ever inserted into `UnstakerByStakeId` map. Is that step missing in `apply_on_curator_opening()`

Reply:
Correction.. (as per chat) step is missing in `deactivate_curator()`

Reply:
Yes this is a bug, we must add to this map whenever there is async unstaking initaited, that is staking with an unboding period.

Reply:
Here is a commit from #103 which makes unstaked a normal method